File: src/solution_java/easy/_121BestTimeToBuyAndSellStock.java
package solution_java.easy;

public class _121BestTimeToBuyAndSellStock {
public class Solution {
    public int maxProfit(int[] prices) {
        int ans = 0;
        int min = Integer.MAX_VALUE;
        for (int price : prices) {
            if (price < min) {
                min = price;
            } else if (price - min > ans) {
                ans = price - min;
            }
        }
        return ans;
    }
}
}

File: src/easy/_121BestTimeToBuyAndSellStock.kt
package easy

class _121BestTimeToBuyAndSellStock {
    class Solution {
        fun maxProfit(prices: IntArray): Int {
            var max = 0
            var minPrice = Int.MAX_VALUE
            for (i in prices.indices) {
                if (prices[i] < minPrice) {
                    minPrice = prices[i]
                } else if (prices[i] - minPrice > max) {
                    max = prices[i] - minPrice
                }
            }
            return max
        }
    }
}